packages/cli-core/src/printer/printer.ts

```diff
@@ -19,10 +20,19 @@ export class Printer {
    */
   constructor(
     private readonly minimumLogLevel: LogLevel,
-    private readonly stdout: NodeJS.WriteStream = process.stdout,
-    private readonly stderr: NodeJS.WriteStream = process.stderr,
+    private readonly stdout:
+      | WriteStream
+      | NodeJS.WritableStream = process.stdout,
+    private readonly stderr:
+      | WriteStream
+      | NodeJS.WritableStream = process.stderr,
     private readonly refreshRate: number = 500
-  ) {}
+  ) {
+    // if not running in tty, turn off colors
```

Comment:
Aren't the methods in `format` already a noop if colors are not enabled? Why do we need additional logic for it here?

Reply:
`format` uses `supportColors` which only looks at the environment/terminal you are running in and decides whether to enable or disable colors. E.g. in CI/CD the colors might be disabled. See https://github.com/isaacs/color-support

In the printer, we might be piping the data in some other place as well (e.g. writing to a file) `supportColors` doesn't know that and will keep the colors enabled.

Reply:
I removed it from printer such that printer will print whatever is given to it. Instead of fiddling with kleur's `$` the caller can then choose to not use `format` or colors.